src/hostname/hostnamed.c

```diff
@@ -218,8 +219,9 @@ static int get_hardware_serial(char **ret) {
         if (r < 0)
                 return log_debug_errno(r, "Failed to open /sys/class/dmi/id device, ignoring: %m");
 
-        (void) sd_device_get_sysattr_value(device, "product_serial", &s);
-        if (isempty(s))
+        if (dmi_data)
```

Comment:
just `assert(dmi_data)` at the top of the function, after the variables declarations, no point in allowing this to be called without a data type